.gitlab/kernel_version_testing/common.yml

```diff
@@ -0,0 +1,285 @@
+  - mkdir junit
+  - mkdir testjson
+  - mkdir pkgjson
+  - popd
+  - pushd $CI_PROJECT_DIR/test/new-e2e/system-probe/test-runner && go build -o $DEPENDENCIES/test-runner && popd
+  - pushd $CI_PROJECT_DIR/test/new-e2e/system-probe/test-json-review && go build -o $DEPENDENCIES/test-json-review && popd
+  - cp $CI_PROJECT_DIR/test/new-e2e/system-probe/test-runner/files/*.json $DEPENDENCIES/
+  # package all the dependencies
+  - ls -la $DEPENDENCIES
+  - pushd $DD_AGENT_TESTING_DIR/$ARCH
+  - tar czvf ../$ARCHIVE_NAME dependencies
+  - popd
+
+# --- Docker test images
+
+.pull_test_dockers:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  needs: []
+  tags: ["arch:amd64"]
+  rules: !reference [.on_any_kmt_test_component_or_manual]
+  stage: kernel_matrix_testing_prepare
+  script:
+    # DockerHub login for build to limit rate limit when pulling base images
+    - DOCKER_REGISTRY_LOGIN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_LOGIN_SSM_KEY --with-decryption --query "Parameter.Value" --out text)
+    - aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.$DOCKER_REGISTRY_PWD_SSM_KEY --with-decryption --query "Parameter.Value" --out text | crane auth login --username "$DOCKER_REGISTRY_LOGIN" --password-stdin "$DOCKER_REGISTRY_URL"
+    # Pull base images
+    - mkdir $KMT_DOCKERS
+    - inv -e system-probe.save-test-dockers --use-crane --output-dir $KMT_DOCKERS --arch $ARCH
+  artifacts:
+    expire_in: 1 day
+    paths:
+      - $KMT_DOCKERS
+  variables:
+    KMT_DOCKERS: $DD_AGENT_TESTING_DIR/kmt-dockers-$ARCH
+
+pull_test_dockers_x64:
+  extends: .pull_test_dockers
+  variables:
+    ARCH: amd64
+
+pull_test_dockers_arm64:
+  extends: .pull_test_dockers
+  variables:
+    ARCH: arm64
+
+# -- Test dependencies
+
+.package_dependencies:
+  stage: kernel_matrix_testing_prepare
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  allow_failure: true
+  before_script:
+    - !reference [.kernel_matrix_testing_new_profile]
+    - !reference [.write_ssh_key_file]
+  tags: ["arch:amd64"]
+  script:
+    # upload dependencies
+    - !reference [.wait_for_instances]
+    - !reference [.get_instance_ip_by_type]
+    - !reference [.setup_ssh_config]
+    - scp $CI_PROJECT_DIR/test/new-e2e/system-probe/test/micro-vm-init.sh metal_instance:/opt/kernel-version-testing
+    - |
+      if [ -d $DD_AGENT_TESTING_DIR/kmt-dockers-$ARCH ]; then
+        tar czvf $DD_AGENT_TESTING_DIR/kmt-dockers-$ARCH.tar.gz $DD_AGENT_TESTING_DIR/kmt-dockers-$ARCH
+        scp $DD_AGENT_TESTING_DIR/kmt-dockers-$ARCH.tar.gz metal_instance:/opt/kernel-version-testing
+      fi
+  variables:
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+
+# -- BTF
+.upload_minimized_btfs:
```

Comment:
Should this be in the common file if these jobs are not used by the security agent?